From 476ce7339868bc676ce7f861357631ca938f723a Mon Sep 17 00:00:00 2001
From: Harsh J <harsh@cloudera.com>
Date: Tue, 17 Mar 2015 00:59:50 +0530
Subject: [PATCH 0816/1023] HDFS-2360. Ugly stacktrce when quota exceeds.
 (harsh)

(cherry picked from commit 046521cd6511b7fc6d9478cb2bed90d8e75fca20)
(cherry picked from commit cbb885836741085757f5e5da3e7413af9cbceaf6)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt
	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java

Change-Id: I0ae400262130b9ad60d9a467fe00e95da840d6f5
(cherry picked from commit ba0da2f5ad4bb06ebb0e53352bb7176e23d4403d)
(cherry picked from commit afe8013e56fa8fbfbb38e9cce597e41dc8ba928a)
---
 .../org/apache/hadoop/hdfs/DFSOutputStream.java    |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
index 30172f7..a8f0891 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
@@ -55,6 +55,7 @@
 import org.apache.hadoop.hdfs.client.HdfsDataOutputStream;
 import org.apache.hadoop.hdfs.client.HdfsDataOutputStream.SyncFlag;
 import org.apache.hadoop.hdfs.protocol.BlockStoragePolicy;
+import org.apache.hadoop.hdfs.protocol.QuotaExceededException;
 import org.apache.hadoop.hdfs.protocol.DSQuotaExceededException;
 import org.apache.hadoop.hdfs.protocol.DatanodeInfo;
 import org.apache.hadoop.hdfs.protocol.ExtendedBlock;
@@ -779,7 +780,13 @@ public void run() {
         } catch (Throwable e) {
           // Log warning if there was a real error.
           if (restartingNodeIndex == -1) {
-            DFSClient.LOG.warn("DataStreamer Exception", e);
+            // Since their messages are descriptive enough, do not always
+            // log a verbose stack-trace WARN for quota exceptions.
+            if (e instanceof QuotaExceededException) {
+              DFSClient.LOG.debug("DataStreamer Quota Exception", e);
+            } else {
+              DFSClient.LOG.warn("DataStreamer Exception", e);
+            }
           }
           if (e instanceof IOException) {
             setLastException((IOException)e);
-- 
1.7.9.5

